### PR TITLE
improve memory footprint of the timer

### LINKED ISSUE #906: Lower memory footprint
Please, look at the series of steps, as a result of which I have reduced the amount of used memory by almost 2 times.

https://github.com/magicprinc/smallrye-fault-tolerance/commits/feature/compact-timer

Plus 
- more correct task-comparator behavior
- support for graceful shutdown in standalone mode (registry of all timers, access to executor)
- monitoring support (registry of all timers, timer queue size)
and other micro cleanups (Collections.newSetFromMap(new ConcurrentHashMap<>())→ConcurrentHashMap.newKeySet())

I hope you find these useful
PR: https://github.com/smallrye/smallrye-fault-tolerance/pull/907

### PR DESCRIPTION
**Pull request by Ladicek**
Fixes #906

### CODE REVIEW

File: implementation/core/src/main/java/io/smallrye/faulttolerance/core/timer/ThreadTimer.java [hunk 1]
```diff
@@ -89,23 +104,25 @@ public ThreadTimer(Executor defaultExecutor) {
                                     try {
```
Comment (magicprinc) [inline implementation/core/src/main/java/io/smallrye/faulttolerance/core/timer/ThreadTimer.java:104]: Unnecessary creation of lambda for each call: Task can implement Runnable and do this without extra allocations.

Comment (Ladicek) [reply implementation/core/src/main/java/io/smallrye/faulttolerance/core/timer/ThreadTimer.java:104]: If we allocate any lambdas anywhere, it's not in the timer. The `schedule` methods already accept a `Runnable`.

Comment (magicprinc) [reply implementation/core/src/main/java/io/smallrye/faulttolerance/core/timer/ThreadTimer.java:104]: GitHub shows here not the lines I have commented. I speak about

`executorForTask.execute(() -> {
    LOG.runningTimerTask(task);
    try { `

Comment (Ladicek) [reply implementation/core/src/main/java/io/smallrye/faulttolerance/core/timer/ThreadTimer.java:104]: Ouch! Understood. Yes, that makes sense, thanks!

Comment (Ladicek) [reply implementation/core/src/main/java/io/smallrye/faulttolerance/core/timer/ThreadTimer.java]: Done.

**Pull request merged**